app/code/Magento/Csp/Helper/InlineUtil.php

```diff
@@ -88,7 +88,7 @@ public function __construct(
      */
     private function generateHashValue(string $content): array
     {
-        return [base64_encode(hash('sha256', $content, true)) => 'sha256'];
+        return [base64_encode(hash('xxh128', $content, true)) => 'sha256'];
```

Comment:
hash algorithm changes in array key, but it's not changed in value, that used for preparing header.

Reply:
I indeed missed this one :/ 
Before editing this, I'll wait for Magento team feedback on which algo we should use here.
Thanks